### PR TITLE
[CDAP-3391] Improve exception/interrupt handling and logging in Abstr…

### PR DESCRIPTION
**Pull request by bdmogal**
…actProgramTwillRunnable and FileLogAppender

Also some additional cleanup of unused code in FileLogAppender.

Jira: [CDAP-3391](https://issues.cask.co/browse/CDAP-3391)
Build: https://builds.cask.co/browse/CDAP-DUT2684

### CODE REVIEW

**Comment by henry-cask** [comment]
The change looks good, but why downgrade to debug level?

**Comment by poornachandra** [comment]
@bdmogal I think any info logging that happens once per run can remain at info level. Any repeated messages can become debug.

**Comment by bdmogal** [comment]
Reverted to info level for one-time logs. Please review again, Thanks!

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/file/FileLogAppender.java [hunk 1]
```diff
@@ -182,6 +176,12 @@ public void stop() {
     }
 
     scheduledExecutor.shutdownNow();
+    try {
+      scheduledExecutor.awaitTermination(Integer.MAX_VALUE, TimeUnit.NANOSECONDS);
```
Comment (poornachandra) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/file/FileLogAppender.java]: Let's wait for a reasonable time, like 5 mins for it to terminate.

**Comment by poornachandra** [comment]
One more comment, please fix and merge.

**Comment by bdmogal** [comment]
Addressed comments. Will merge after next successful build - http://builds.cask.co/browse/CDAP-DUT2684-3

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java [hunk 2]
```diff
@@ -313,11 +316,11 @@ public void error(Throwable cause) {
 
     try {
       state.get();
-      LOG.info("Program stopped.");
+      LOG.info("Program {} stopped.", name);
     } catch (InterruptedException e) {
-      LOG.warn("Program interrupted.", e);
+      LOG.warn("Program {} interrupted.", name, e);
```
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java]: Should we set the interrupt flag here as well?

Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java]: Added

**Comment by bdmogal** [comment]
Tested on cluster. @sagarkapare please take another look.

**Comment by sagarkapare** [comment]
LGTM :+1:

**Comment by bdmogal** [comment]
Squashed commits. Build is green. Integration tests pass. Merging.

**Pull request merged**